pkg/networkpolicy/multinetworkpolicy.go

```diff
@@ -0,0 +1,328 @@
+package networkpolicy
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/k8snetworkplumbingwg/multi-networkpolicy/pkg/apis/k8s.cni.cncf.io/v1beta1"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// MultiNetworkPolicyBuilder provides struct for MultiNetworkPolicy object.
+type MultiNetworkPolicyBuilder struct {
+	// MultiNetworkPolicy definition. Used to create MultiNetworkPolicy object with minimum set of required elements.
+	Definition *v1beta1.MultiNetworkPolicy
+	// Created MultiNetworkPolicy object on the cluster.
+	Object *v1beta1.MultiNetworkPolicy
+	// api client to interact with the cluster.
+	apiClient *clients.Settings
+	// errorMsg is processed before MultiNetworkPolicy object is created.
+	errorMsg string
+}
+
+// NewMultiNetworkPolicyBuilder method creates new instance of builder.
+func NewMultiNetworkPolicyBuilder(apiClient *clients.Settings, name, nsname string) *MultiNetworkPolicyBuilder {
+	glog.V(100).Infof(
+		"Initializing new MultiNetworkPolicyBuilder structure with the following params: name: %s, namespace: %s",
+		name, nsname)
+
+	builder := &MultiNetworkPolicyBuilder{
+		apiClient: apiClient,
+		Definition: &v1beta1.MultiNetworkPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the MultiNetworkPolicy is empty")
+
+		builder.errorMsg = "The MultiNetworkPolicy 'name' cannot be empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The namespace of the MultiNetworkPolicy is empty")
+
+		builder.errorMsg = "The MultiNetworkPolicy 'namespace' cannot be empty"
+	}
+
+	return builder
+}
+
+// WithPodSelector adds podSelector to MultiNetworkPolicy.
+func (builder *MultiNetworkPolicyBuilder) WithPodSelector(
+	podSelector *metav1.LabelSelector) *MultiNetworkPolicyBuilder {
```

Comment:
Is there a reason you are using *metav1.LabelSelector here? If you just use metav1.LabelSelector, you will not need to check if it is nil nor dereference it on line 80